pkg/api/register.go

```diff
@@ -111,6 +111,15 @@ func AddToScheme(scheme *runtime.Scheme) {
 		&unversioned.APIGroup{},
 		&unversioned.APIResourceList{},
 	)
+
+	// Types registered here will be kept unversioned when they are encoded.
+	// It's for backward compatibility with release 1.1. clients.
+	Scheme.AddToKeepUnverioned(
```

Comment:
We need to control serialization on the apiserver and deserialization on client impls.  For client impls, we want to tolerate both paths (which means we need these types registered in an API group as well as outside).  To control server side, I think it has to be something that we know about when we render them in the discovery handlers.  The handlers are rendered outside of a group today.  We would need to have a specific codec wrapper for stripping api group version info, and then use that particular codec in negotiate only for certain calls.